internal/elasticsearch/certificates.go

```diff
@@ -557,8 +556,15 @@ func ensureSessionSecret(secretLength int, runes []rune, session *sessionSecret)
 
 	// otherwise its different, regenerate it
 	b := make([]rune, secretLength)
+	bigRuneCount := big.NewInt(int64(len(runes)))
+
 	for i := range b {
-		b[i] = runes[mathRand.Intn(len(runes))]
+		bigIndex, err := rand.Int(rand.Reader, bigRuneCount)
+		if err != nil {
+			return err
+		}
+
+		b[i] = runes[int(bigIndex.Int64())]
```

Comment:
What does this change have in common with updating the sdk? Is this more a Golang 1.18 thing?

Reply:
This is a golangci lint error for golang 1.18 as far aas I'm aware. I can break this into two PRs if needed.

Reply:
No it's ok here